code/data_structures/src/list/singly_linked_list/operations/find/search_element_python.py

```diff
@@ -0,0 +1,103 @@
+class Node:
+    '''
+    A Node in a singly-linked list
+
+    Parameters
+    -------------------
+    data :
+        The data to be stored in the node
+    next:
+        The link to the next node in the singly-linked list
+    '''
+
+    def __init__(self, data=None, next=None):
+        self.data = data
+        self.next = next
+
+    def __repr__(self):
+        ''' Node representation as required'''
+        return (self.data)
+
+class NodeAdapter1(Node):
+    def __bool__(self):
+        return False
+
+class NodeAdapter2(Node):
+    def __bool__(self):
+        return True
+    
+class SinglyLinkedList:
+    def __init__(self,NodeType):
+        self.head = None
+        self._NodeType = NodeType
+
+    def append(self, data):
+        '''
+        Inserts New data at the ending of the Linked List
+        Takes O(n) time
+        '''
+        new_node = _NodeType(data = data)
```

Comment:
Miss `self.` here.

Reply:
Don't use spaces around the = sign when used to indicate a keyword argument or a default parameter value. [\<li> 4th.](https://www.python.org/dev/peps/pep-0008/?#other-recommendations)